sqlglot/dialects/athena.py

```diff
@@ -122,8 +145,61 @@ class Parser(Trino.Parser):
         STATEMENT_PARSERS = {
             **Trino.Parser.STATEMENT_PARSERS,
             TokenType.USING: lambda self: self._parse_as_command(self._prev),
+            TokenType.SHOW: lambda self: self._parse_show(),
         }
 
+        SHOW_PARSERS = {
+            "COLUMNS FROM": _show_parser("COLUMNS", target="FROM"),
+            "COLUMNS IN": _show_parser("COLUMNS", target="IN"),
+            "CREATE TABLE": _show_parser("CREATE TABLE", target=True),
+            "CREATE VIEW": _show_parser("CREATE VIEW", target=True),
+            "DATABASES": _show_parser("DATABASES"),
+            "PARTITIONS": _show_parser("PARTITIONS", target=True),
+            "SCHEMAS": _show_parser("SCHEMAS"),
+            "TABLES": _show_parser("TABLES", target="IN"),
+            "TBLPROPERTIES": _show_parser("TBLPROPERTIES", target=True),
+            "VIEWS": _show_parser("VIEWS", target="IN"),
+        }
+
+        def _parse_show_athena(
+            self,
+            this: str,
+            target: bool | str = False,
+        ) -> exp.Show:
+            if target:
+                if isinstance(target, str):
+                    self._match_text_seq(target)
+                target_id = self._parse_id_var()
+            else:
+                target_id = None
+
+            db = None
+
+            if self._match(TokenType.FROM) or self._match(TokenType.IN):
+                db = self._parse_id_var()
+            elif self._match(TokenType.DOT):
+                db = target_id
+                target_id = self._parse_id_var()
+
+            # using `rlike` for the specific commands where athena uses regex in a `LIKE`
+            like = (self._match_text_seq("LIKE") and self._parse_string()) or None
+            rlike = None
```

Comment:
Why is this necessary? Can't we just leave the `Like` node as is without introducing a new argument in the AST?